crates/ty_python_semantic/src/module_resolver/list.rs

```diff
@@ -0,0 +1,1688 @@
+            list_snapshot_filter(&db, |m| m.name(&db).as_str().contains("pydoc_data")),
+            @r#"
+        [
+            Module::File("pydoc_data", "std-vendored", "stdlib/pydoc_data/__init__.pyi", Package, None),
+        ]
+        "#,
+        );
+    }
+
+    #[test]
+    fn resolve_package() {
+        let TestCase { db, .. } = TestCaseBuilder::new()
+            .with_src_files(&[("foo/__init__.py", "print('Hello, world!'")])
+            .build();
+
+        insta::assert_debug_snapshot!(
+            list_snapshot(&db),
+            @r#"
+        [
+            Module::File("foo", "first-party", "//src/foo/__init__.py", Package, None),
+        ]
+        "#,
+        );
+    }
+
+    #[test]
+    fn package_priority_over_module() {
+        const SRC: &[FileSpec] = &[
+            ("foo/__init__.py", "print('Hello, world!')"),
+            ("foo.py", "print('Hello, world!')"),
+        ];
+
+        let TestCase { db, .. } = TestCaseBuilder::new().with_src_files(SRC).build();
+
+        insta::assert_debug_snapshot!(
+            list_snapshot(&db),
+            @r#"
+        [
+            Module::File("foo", "first-party", "//src/foo/__init__.py", Package, None),
+        ]
+        "#,
+        );
+    }
+
+    #[test]
+    fn typing_stub_over_module() {
+        const SRC: &[FileSpec] = &[("foo.py", "print('Hello, world!')"), ("foo.pyi", "x: int")];
+
+        let TestCase { db, .. } = TestCaseBuilder::new().with_src_files(SRC).build();
+
+        insta::assert_debug_snapshot!(
+            list_snapshot(&db),
+            @r#"
+        [
+            Module::File("foo", "first-party", "//src/foo.pyi", Module, None),
+        ]
+        "#,
+        );
+    }
+
+    #[test]
+    fn sub_packages() {
+        const SRC: &[FileSpec] = &[
+            ("foo/__init__.py", ""),
+            ("foo/bar/__init__.py", ""),
+            ("foo/bar/baz.py", "print('Hello, world!)'"),
+        ];
+
+        let TestCase { db, .. } = TestCaseBuilder::new().with_src_files(SRC).build();
+
+        insta::assert_debug_snapshot!(
+            list_snapshot(&db),
+            @r#"
+        [
+            Module::File("foo", "first-party", "//src/foo/__init__.py", Package, None),
+        ]
+        "#,
+        );
+    }
+
+    #[test]
+    fn module_search_path_priority() {
+        let TestCase { db, .. } = TestCaseBuilder::new()
+            .with_src_files(&[("foo.py", "")])
+            .with_site_packages_files(&[("foo.py", "")])
+            .build();
+
+        insta::assert_debug_snapshot!(
+            list_snapshot(&db),
+            @r#"
+        [
+            Module::File("foo", "first-party", "//src/foo.py", Module, None),
+        ]
+        "#,
+        );
+    }
+
+    #[test]
+    #[cfg(target_family = "unix")]
+    fn symlink() -> anyhow::Result<()> {
+        use anyhow::Context;
+
+        let mut db = TestDb::new();
+
+        let temp_dir = tempfile::TempDir::with_prefix("PREFIX-SENTINEL")?;
```

Comment:
Nit: You can change the underlying system of a `TestDb` by using `db.use_system(other)`. See this test for an example https://github.com/astral-sh/ruff/blob/f34b65b7a00dfc61c5a54e4280b421c94151e677/crates/ty_python_semantic/src/module_resolver/resolver.rs#L2158-L2164

It's up to you if you think this simplifies anything or not.

Reply:
Is this what you mean? https://github.com/astral-sh/ruff/pull/19883/files#r2273636330